lib/components/admin/print-field-trip-layout.js

```diff
@@ -0,0 +1,235 @@
+import PrintableItinerary from '@opentripplanner/printable-itinerary'
+import React, { Component } from 'react'
+import { Button } from 'react-bootstrap'
+import { connect } from 'react-redux'
+import styled from 'styled-components'
+
+import * as callTakerActions from '../../actions/call-taker'
+import * as fieldTripActions from '../../actions/field-trip'
+import TripDetails from '../narrative/connected-trip-details'
+import { getTripFromRequest, lzwDecode } from '../../util/call-taker'
+import { ComponentContext } from '../../util/contexts'
+import { getTitle } from '../../util/state'
+
+// Styles specific for rendering PrintFieldTripLayout.
+const PrintLayout = styled.div`
+  font-size: 16px;
+  line-height: 115%;
+  margin: 8px;
+`
+
+const Header = styled.div``
+
+const TripTitle = styled.h1`
+  border-bottom: 3px solid gray;
+  font-size: 30px;
+  font-weight: bold;
+`
+
+const TripInfoList = styled.ul`
+  font-size: 16px;
+  list-style: none;
+  margin-top: 1em;
+  padding: 0;
+`
+
+export const Val = styled.span`
+  :empty:before {
+    content: 'N/A';
+  }
+`
+
+// The styles below mirror those found in OTP native client.
+const TripContainer = styled.div`
+  background: #ddd;
+  margin-top: 1em;
+
+  & > h2 {
+    font-size: 20px;
+    font-weight: bold;    
+    margin: 0;
+    padding: 4px;
+  }
+`
+
+const TripBody = styled.div`
+  padding: 8px;
+`
+
+const ItineraryContainer = styled.div`
+  border: 3px solid #444;
+  margin-top: .5em;
+
+  & > h3 {
+    background: #444;
+    color: white;    
+    font-size: 18px;
+    font-weight: bold;
+    margin: 0;
+    padding: 4px;
+  }
+`
+
+const ItineraryBody = styled.div`
+  background: white;
+  padding: 12px;
+`
+
+const TripSummary = styled(TripDetails)`
+  background: #eee;
+  border: 1px solid #bbb;
+  border-radius: 0;
+  margin-top: 15px;
+  padding: 5px;
+`
+
+/**
+ * Component that renders the print version of field trip itineraries.
+ */
+class PrintFieldTripLayout extends Component {
+  static contextType = ComponentContext
+
+  _print = () => {
+    window.print()
+  }
+
+  componentDidMount () {
+    const { initializeModules, title } = this.props
+    // Add print-view class to html tag to ensure that iOS scroll fix only applies
+    // to non-print views.
+    const root = document.getElementsByTagName('html')[0]
+    root.setAttribute('class', 'print-view')
+
+    // Set window title (appears in print headings)
+    document.title = title
+
+    // Load call-taker/field-trip functionality (performs a fetch).
+    initializeModules()
+  }
+
+  async componentDidUpdate (prevProps) {
+    const { fetchFieldTrips, fetchFieldTripDetails, requestId, session } = this.props
+    if (!prevProps.session && session) {
+      // When session is set, load all field trips,
+      // then load details for field trip per request id.
+      await fetchFieldTrips()
```

Comment:
Trimet has thousands of field trips in its database. Is it possible to load a single field trip by request id?

Reply:
We can probably cheat around `fetchFieldTrips`.

Reply:
Good suggestion, updated in https://github.com/opentripplanner/otp-react-redux/pull/398/commits/9e3a9ea50bb4c662aec3aa8d0565d822daf81196.